src/OmniSharp.Roslyn.CSharp/Services/Navigation/GotoDefinitionService.cs

```diff
@@ -47,6 +47,8 @@ public GotoDefinitionService(OmniSharpWorkspace workspace, MetadataHelper metada
                 // go to definition for namespaces is not supported
                 if (symbol != null && !(symbol is INamespaceSymbol))
                 {
+                    if (symbol is IMethodSymbol methodSymbol && methodSymbol.AssociatedSymbol is IPropertySymbol)
```

Comment:
there is already the same `is IMethodSymbol` 2 lines lower so you may want to roll them up together and just check for `IPropertySymbol` there